### PR TITLE
Stop validating binding/template in Triggers

### PR DESCRIPTION
**Pull request by dibyom**
# Changes

Currently, while validating triggers in an EventListener, we fetch
all the bindings and templates, and validate that they exist. This
means that the bindings and the templates have to be created before
creating EventListeners leading to the fragile object creation order
issues listed in #187

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```
We no longer validate the existence of `TriggerBindings` and `TriggerTemplates` when validating a EventListener.

```

### CODE REVIEW

**Comment by dibyom** [comment]
Looks like coverage has dropped off a bit :thinking:

File: pkg/apis/triggers/v1alpha1/event_listener_validation_test.go [hunk 1]
```diff
@@ -4,26 +4,84 @@ import (
...
+		name: "Valid EventListener No Interceptor",
+		el: bldr.EventListener("name", "namespace",
+			bldr.EventListenerSpec(
+				bldr.EventListenerTrigger("tb", "tt", "v1alpha1"))),
+	}, {
+		name: "Valid EventListener Interceptor Name only",
+		el: bldr.EventListener("name", "namespace",
+			bldr.EventListenerSpec(
+				bldr.EventListenerTrigger("tb", "tt", "v1alpha1",
+					bldr.EventListenerTriggerInterceptor("svc", "", "", "")))),
```
Comment (bobcatfish) [inline pkg/apis/triggers/v1alpha1/event_listener_validation_test.go]: im confused why it is valid to only specify the name of an interceptor, is that enough to be able to address it?

File: pkg/apis/triggers/v1alpha1/event_listener_validation_test.go [hunk 2]
```diff
@@ -4,26 +4,84 @@ import (
...
+				bldr.EventListenerTrigger("tb", "tt", "v1alpha1",
+					bldr.EventListenerTriggerInterceptor("svc", "v1", "Service", "namespace"),
+				),
+				bldr.EventListenerTrigger("tb", "tt", "v1alpha1"))),
+	},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			err := test.el.Validate(context.TODO())
```
[the remark this answers is not in the document]
Comment (dibyom) [reply pkg/apis/triggers/v1alpha1/event_listener_validation_test.go]: yeap, good call!

File: pkg/apis/triggers/v1alpha1/event_listener_validation_test.go (hunk 1 above)
Comment (dibyom) [reply pkg/apis/triggers/v1alpha1/event_listener_validation_test.go]: Good catch. Not sure why the validation was done this way but we did return an error later when we actually try to resolve it to a URL.

Comment (dibyom) [reply pkg/apis/triggers/v1alpha1/event_listener_validation_test.go]: Made this an error test!

Comment (bobcatfish) [reply pkg/apis/triggers/v1alpha1/event_listener_validation_test.go]: wooooot :D

**Pull request merged**